### PR TITLE
Add manual e2e testing

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Towards https://github.com/giantswarm/roadmap/issues/3616

This PR adds the manual e2e test procedure and the related script to check that the logging-operator behaves as expected in a GS environment.

Tested and working on `grizzly`.

### CODE REVIEW

File: tests/manual_e2e/basic_test.sh [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+# Helper function - clean up the WC
+clean_wc() {
+  kubectl delete -f grizzly-e2e-wc.yaml
+  rm grizzly-e2e-wc.yaml
+}
+
+# Helper function - checks the status of the daemonset
+check_daemonset_status() {
+  local daemonsetDesiredReplicas=$(kubectl get daemonset -n kube-system --context teleport.giantswarm.io-$1-loggingoperatortest $2 -o yaml | yq .status.desiredNumberScheduled)
```
Comment (hervenicol) [inline tests/manual_e2e/basic_test.sh:17]: image

...please :pray:

**Review by hervenicol — COMMENTED** [review]
Could you please run `shellcheck` on your script(s)?
I'll have a better review after that :)

File: tests/manual_e2e/basic_test.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+{{end}}' | sed -ne 's/false\/aws-//p' | sort -V | tail -1)"
+
+  # Creating WC template and applying it
+  kubectl gs template cluster --provider capa --name loggingoperatortest --organization giantswarm --description "logging-operator e2e tests" --release "$toUseRelease" > grizzly-e2e-wc.yaml
+  kubectl create -f grizzly-e2e-wc.yaml
+
+  echo "WC named 'loggingoperatortest' created. Waiting for it to be ready"
+
+  # Waiting for 1min for the cluster resource to be created
+  sleep 120
```
Comment (hervenicol) [inline tests/manual_e2e/basic_test.sh:63]: Rather have no duration announced in the comment than an outdated one.

```suggestion
  # Allowing a bit of time for the cluster resource to be created
  sleep 120
```

File: tests/manual_e2e/basic_test.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  echo "Checking if the logging agent is up and running on the WC"
+
+  # Logging into the WC to get the context into the kubeconfig
+  tsh kube login "$1"-loggingoperatortest
+  tsh kube login "$1"
+
+  promtail="$(kubectl get apps -n org-giantswarm | grep loggingoperatortest-promtail)"
+  alloy="$(kubectl get apps -n org-giantswarm | grep loggingoperatortest-alloy-logs)"
+
+  if [[ -n "$promtail" ]]; then
```
Comment (hervenicol) [inline tests/manual_e2e/basic_test.sh:79]:
```suggestion
  # We need either promtail or alloy to be running on the workload cluster
  if [[ -n "$promtail" ]]; then
```

**Review by hervenicol — COMMENTED** [review]
Globally I think the script is missing comments, and it's not clear what it does / expects.
Also, error outputs are not clear.
Maybe prefixing them with `[ERROR]` would help?

**Comment by QuantumEnigmaa** [comment]
@hervenicol I did some minor changes concerning the comments and error messages. WDYT ? 
I can't see where I could add more comments as IMO when comments are missing, the output's messages or variables' names are clear enought to explain what's done in the code.

File: tests/manual_e2e/basic_test.sh [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    kubectl wait -n org-giantswarm --for=jsonpath='{.status.release.status}'=deployed app/loggingoperatortest-promtail --timeout=10m
+    sleep 120 # Giving extra time for the daemonset's pods to be ready
+    check_daemonset_status "$1" "promtail"
+  elif [[ -n "$alloy" ]]; then
+    kubectl wait -n org-giantswarm --for=jsonpath='{.status.release.status}'=deployed app/loggingoperatortest-alloy-logs --timeout=10m
+    sleep 120 # Giving extra time for the daemonset's pods to be ready
+    check_daemonset_status "$1" "alloy-logs"
+  else
+    echo "No logging agent app found. Cleaning the WC"
```
Comment (hervenicol) [inline tests/manual_e2e/basic_test.sh:89]: The space before the semicolon is very French :)
But keeping it consistent here.
```suggestion
    echo "Error : No logging agent app found. Cleaning the WC"
```

**Comment by hervenicol** [comment]
> @hervenicol I did some minor changes concerning the comments and error messages. WDYT ? I can't see where I could add more comments as IMO when comments are missing, the output's messages or variables' names are clear enought to explain what's done in the code.

Well, there's something I don't really like in the script's structure. But I can't really put the finger on it or how to improve it. Just a feeling that it should be more straightforward to read given what it does.

Not a reason to block it though. Fine with me.

**Review by hervenicol — APPROVED** [review]
Fine with me.

**Comment by QuantumEnigmaa** [comment]
I'll merge it for now but if you ever find what was troubling you, please go ahead and make the necessary changes to the script :)

**Pull request merged**